lib/chef/provider/package/powershell.rb

```diff
@@ -0,0 +1,136 @@
+          end
+        end
+
+        def remove_package(names, versions)
+          # Removes the package, if no version is passed, all installed version  will be removed
+          names.each_with_index do |name, index|
+            if !versions.nil?
+              if versions[index] != nil
+                powershell_out( "Uninstall-Package #{name} -Force -RequiredVersion #{versions[index]}", { :timeout => @new_resource.timeout })
+              else
+                powershell_out( "Uninstall-Package #{name} -Force", { :timeout => @new_resource.timeout })
+              end
+            end
+            if new_resource.version.nil?
+              stdout = 0
+              while stdout != ""
+                stdout = powershell_out( "Uninstall-Package #{name} -Force", { :timeout => @new_resource.timeout }).stdout
+                if !stdout.empty?
+                  stdout = stdout.split(" ")
+                  Chef::Log.debug("Removed package #{name} with version #{stdout[stdout.index(name) + 1]}")
+                end
+              end
+            end
+          end
+        end
+
+        # return array of latest available packages online
+        def build_candidate_versions
+          versions = []
+          new_resource.package_name.each_with_index do |name, index|
+            if !new_resource.version.nil?
+              if new_resource.version[index] != nil
+                stdout = powershell_out("Find-Package #{name} -RequiredVersion #{new_resource.version[index]}", { :timeout => @new_resource.timeout }).stdout
+              else
+                stdout = powershell_out("Find-Package #{name}", { :timeout => @new_resource.timeout }).stdout
+              end
+            else
+              stdout = powershell_out("Find-Package #{name}", { :timeout => @new_resource.timeout }).stdout
+            end
+            versions.push(find_version(stdout))
+          end
+          versions
+        end
+
+        #return array of currently installed version
+        def build_current_versions
+          version_list = []
+          new_resource.package_name.each_with_index do |name, index|
+            if !new_resource.version.nil?
+              if new_resource.version[index] != nil
+                stdout = powershell_out("Get-Package -Name #{name} -RequiredVersion #{new_resource.version[index]}", { :timeout => @new_resource.timeout }).stdout
+              else
+                stdout = powershell_out("Get-Package -Name #{name}", { :timeout => @new_resource.timeout }).stdout
+              end
+            else
+              stdout = powershell_out("Get-Package -Name #{name}", { :timeout => @new_resource.timeout }).stdout
+            end
+            if stdout.empty?
+              version_list.push(nil)
+            else
+              stdout = stdout.split(" ")
+              version_list.push(stdout[stdout.index(name) + 1])
+            end
+          end
+          version_list
+        end
+
+        def find_version(stdout)
```

Comment:
what are we really passing to `find_version`? Can we name `stdout` something more appropriate? it's overused in this file. maybe this method would be more aptly named `parse_version` and the argument is just a string and could be named string or output?